kayenta-oss/src/main/java/com/netflix/kayenta/oss/storage/OssStorageService.java

```diff
@@ -0,0 +1,464 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.kayenta.oss.storage;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSException;
+import com.aliyun.oss.model.*;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.netflix.kayenta.alicloud.security.AliCloudNamedAccountCredentials;
+import com.netflix.kayenta.canary.CanaryConfig;
+import com.netflix.kayenta.index.CanaryConfigIndex;
+import com.netflix.kayenta.index.config.CanaryConfigIndexAction;
+import com.netflix.kayenta.security.AccountCredentialsRepository;
+import com.netflix.kayenta.storage.ObjectType;
+import com.netflix.kayenta.storage.StorageService;
+import com.netflix.kayenta.util.Retry;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.time.Instant;
+import java.util.*;
+import javax.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Singular;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.StringUtils;
+
+@Builder
+@Slf4j
+public class OssStorageService implements StorageService {
+
+  public final int MAX_RETRIES = 10;
+  public final long RETRY_BACKOFF = 1000;
+
+  @NotNull private ObjectMapper objectMapper;
+
+  @NotNull @Singular @Getter private List<String> accountNames;
+
+  @Autowired AccountCredentialsRepository accountCredentialsRepository;
+
+  @Autowired CanaryConfigIndex canaryConfigIndex;
+
+  @Override
+  public boolean servicesAccount(String accountName) {
+    return accountNames.contains(accountName);
+  }
+
+  private final Retry retry = new Retry();
+
+  /** Check to see if the bucket exists, creating it if it is not there. */
+  public void ensureBucketExists(String accountName) {
+    AliCloudNamedAccountCredentials credentials =
+        (AliCloudNamedAccountCredentials)
+            accountCredentialsRepository
+                .getOne(accountName)
+                .orElseThrow(
+                    () ->
+                        new IllegalArgumentException(
+                            "Unable to resolve account " + accountName + "."));
+
+    OSS ossClient = credentials.getOssClient();
+    String bucket = credentials.getBucket();
+    CreateBucketRequest createBucketRequest = new CreateBucketRequest(bucket);
+
+    try {
+      ossClient.createBucket(createBucketRequest);
+    } catch (OSSException oe) {
+      log.error(
```

Comment:
Should this throw the error, is the application healthy if someone has this enabled but it can't create the bucket?